aws/usageReports/es/monthlyReport.go

```diff
@@ -0,0 +1,166 @@
+			return err
+		}
+		stats, err := getDomainStats(ctx, *domain.DomainName, sess, start, end)
+		if err != nil {
+			logger.Error("Error while getting Domain stats", err.Error())
+			return err
+		}
+		logger.Info(fmt.Sprintf("Domain report : \n%+#v\nStats : \n%+#v\n", domain, stats), nil)
+		detail := make(map[string]float64, 0)
+		detail["domain"] = domainCost.Cost
+		domainChan <- Domain{
+			Arn:               aws.StringValue(domain.ARN),
+			InstanceType:      aws.StringValue(domain.ElasticsearchClusterConfig.InstanceType),
+			InstanceCount:     aws.Int64Value(domain.ElasticsearchClusterConfig.InstanceCount),
+			DomainID:          aws.StringValue(domain.DomainId),
+			DomainName:        aws.StringValue(domain.DomainName),
+			TotalStorageSpace: aws.Int64Value(domain.EBSOptions.VolumeSize),
+			Region:            region,
+			Tags:              getDomainTag(tags.TagList),
+			CPUUtilizationAverage:    stats.CPUUtilizationAverage,
+			CPUUtiliztionPeak:        stats.CPUUtiliztionPeak,
+			FreeStorageSpace:         stats.FreeStorageSpace,
+			JVMMemoryPressureAverage: stats.JVMMemoryPressureAverage,
+			JVMMemoryPressurePeak:    stats.JVMMemoryPressurePeak,
+			Cost:       domainCost.Cost,
+			CostDetail: detail,
+		}
+	}
+	return nil
+}
+
+// getEsMetrics gets credentials, accounts and region to fetch RDS instances stats
+func getEsMetrics(ctx context.Context, domains []utils.CostPerResource, aa taws.AwsAccount, startDate, endDate time.Time) (Report, error) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	creds, err := taws.GetTemporaryCredentials(aa, ESStsSessionName)
+	if err != nil {
+		logger.Error("Error when getting temporary credentials", err.Error())
+		return Report{}, err
+	}
+	defaultSession := session.Must(session.NewSession(&aws.Config{
+		Credentials: creds,
+		Region:      aws.String(config.AwsRegion),
+	}))
+	account, err := utils.GetAccountId(ctx, defaultSession)
+	if err != nil {
+		logger.Error("Error when getting account id", err.Error())
+		return Report{}, err
+	}
+	report := Report{
+		Account:    account,
+		ReportDate: startDate,
+		ReportType: "monthly",
+		Domains:    make([]Domain, 0),
+	}
+	regions, err := utils.FetchRegionsList(ctx, defaultSession)
+	if err != nil {
+		logger.Error("Error when fetching regions list", err.Error())
+		return Report{}, err
+	}
+	domainChans := make([]<-chan Domain, 0, len(regions))
+	for _, region := range regions {
+		domainChan := make(chan Domain)
+		go fetchMonthlyDomainsList(ctx, creds, domains, region, domainChan, startDate, endDate)
+		domainChans = append(domainChans, domainChan)
+	}
+	for domain := range merge(domainChans...) {
+		report.Domains = append(report.Domains, domain)
+	}
+	return report, nil
+}
+
+func filterEsDomains(esCost []utils.CostPerResource) []utils.CostPerResource {
+	costDomains := []utils.CostPerResource{}
+	for _, domain := range esCost {
+		split := strings.Split(domain.Resource, ":")
+		if len(split) == 6 && split[2] == "es" {
```

Comment:
this is a magic condition, maybe could you comment why do you use this condition